src/index.spec.ts

```diff
@@ -324,11 +366,6 @@ describe(`partial`, () => {
   it(`allows "empty" object`, () => {
     expect(subType({ a: undefined, b: undefined, c: undefined })).to.equal(true)
   })
-  it(`doesn't allow random stuff like arrays of numbers`, () => {
-    expect(subType([])).to.equal(false, `Array`)
-    expect(subType(0)).to.equal(false, `Number`)
-    expect(subType('a')).to.equal(false, `String`)
-  })
```

Comment:
This test breaks the typing of `partial` so I removed it.

If handling the case of invalid types is wanted then `partial` should be updated to something like this: 

```ts
export function partial<T extends Dict> (guard: GuardWithShape<T>): GuardWithShape<Partial<T>>
export function partial(guard: unknown): (input: unknown) => false
export function partial<T>(guard: unknown): Function {
  if (typeof guard !== 'object' || guard === null) {
    return () => false;
  }
  const safeGuard = guard as GuardWithShape<T>;
  const resultShape: any = {}
  for (const key of Object.keys(safeGuard.shape)) {
    resultShape[key] = isOneOf(isUndefined, safeGuard.shape[key] as any)
  }
  return isOfShape(resultShape, safeGuard.exact) as any
}
```